ts/components/messages/MessageAgenda.tsx

```diff
@@ -236,12 +266,73 @@ const FakeItemComponent = (
  */
 class MessageAgenda extends React.PureComponent<Props, State> {
   private sectionListRef = React.createRef<any>();
+  private overScrollAnim: Animated.Value;
+
+  public componentDidMount() {
+    this.state.scrollY.addListener(value => this.handleScroll(value));
+  }
+
+  public componentWillUnmount() {
+    this.state.scrollY.removeAllListeners();
+  }
+
+  public handleRelease() {
+    if (pot.isLoading(this.state.potAnim) && !pot.isSome(this.state.potAnim)) {
+      // start animation
+      this.setState({ potAnim: pot.some(this.state.potAnim) });
+      this.animateOverScroll();
+    }
+  }
+
+  private handleScroll(pullDownDistance: any) {
+    if (pullDownDistance.value <= MIN_PULLDOWN_DISTANCE) {
+      return this.setState({ potAnim: pot.toLoading(this.state.potAnim) });
+    } else {
+      this.animateCloseOverScroll();
+      return this.setState({ potAnim: pot.none });
+    }
+  }
+
+  private handleDragMove() {
+    if (pot.isLoading(this.state.potAnim) && !pot.isSome(this.state.potAnim)) {
```

Comment:
```suggestion
    if (pot.isLoading(this.state.potAnim)) {
```
if pot is loading it can't be some

Reply:
@Undermaken  Why you think isLoading pot can't be some? isLoading check the union of PotNoneLoading and PotSomeLoading. Pot can also be in "updating" state so with a value but still loading/checking for new value.

Reply:
@fpersico you are right. I just check **isLoading** checks if it is **noneLoading** and **someLoading** too
Actually in this code someLoading is never used so I guess @Edigio can avoid the second check in the if statement. Are you agree?